packages/mongodb-constants/src/atlas-search-templates.ts

```diff
@@ -0,0 +1,120 @@
+import dedent from 'dedent';
+
+export type SearchTemplate = {
+  name: string;
+  snippit: string;
+  version: string;
+};
+
+export const ATLAS_SEARCH_TEMPLATES: SearchTemplate[] = [
+  {
+    name: 'Dynamic field mappings',
+    snippit: dedent`{
+            "mappings": {
+              "dynamic": true
+            }
+        }`,
+    version: '4.4.0',
+  },
+  {
+    name: 'Static field mappings',
+    snippit: dedent`{
+            "mappings": {
+              "dynamic": false,
+              "fields": {
+                <string field name>: {
+                    "type": "string"
+                },
+                <number field name>: {
+                    "type": "number"
+                },
+                <date field name>: {
+                    "type": "date"
+                },
+                <geo field name>: {
+                    "type": "geo"
+                },
```

Comment:
When this snippet is applied in the cm editor, user should be able to use `tab` to navigate to placeholder texts (in this case string|number|date|geo field names. We may have to do something like [this](https://github.com/mongodb-js/devtools-shared/blob/db3b603b83bf47643e48fedb4fc6e4f3d2261c83/packages/mongodb-constants/src/stage-operators.ts#L81)